addons/stock/models/stock_inventory.py

```diff
@@ -131,7 +131,7 @@ def post_inventory(self):
         # The inventory is posted as a single step which means quants cannot be moved from an internal location to another using an inventory
         # as they will be moved to inventory loss, and other quants will be created to the encoded quant location. This is a normal behavior
         # as quants cannot be reuse from inventory location (users can still manually move the products before/after the inventory if they want).
-        self.mapped('move_ids').filtered(lambda move: move.state != 'done')._action_done()
+        self.env['stock.move'].search([('inventory_id', 'in', self.ids), ('state', '!=', 'done')])._action_done()
```

Comment:
It is really useful ? the gain is minimal, no ?

Reply:
Yeah it's quite small. I added it because it was in the execution flow of the inventory validation that I was optimizing. If we want only minimal, impactful changes then it can safely be dropped without changing the performance gains reported above.